include/grpcpp/server.h

```diff
@@ -338,6 +347,12 @@ class Server : public ServerInterface, private internal::GrpcLibrary {
   // Shutdown.  Even though this is only used with NDEBUG, instantiate it in all
   // cases since otherwise the size will be inconsistent.
   std::vector<CompletionQueue*> cq_list_;
+
+  // Whetner per-call load reporting is enabled.
+  bool call_metric_recording_enabled_ = false;
```

Comment:
The server metric recorder is optional for cases where the service only needs per-call recording, as in RR + GSLB in prod. FYI I summarized cases in the last table in [go/grpc-backend-metrics-update](http://go/grpc-backend-metrics-update) and my draft prod API exposes an option for the service to disable this and the backend metric sampling.

I think we don't necessarily need to require ServerMetricRecorder here unlike OrcaService. Let me know what you think.

Reply:
I'll update once can conclude what we want:

- Creating CMR lazily makes a lot of sense and I can update. I just kept the old behavior where it's always created when call metric recording is enabled (OrcaInterceptor creates upon seeing the init metadata).
- But I also want to keep the current behavior that GetCallMetricRecorder() returns null when call metric recording is not enabled (effectively, since OrcaInterceptor creates it). I think this is a very good design, it will save users from misusing and save time to debug as they will fail outright instead of silently falling back to RR.
- Current implementation already is designed so that if SMR is set, CMR is always set. The case I raise above is the inverse. When CMR is set and the call metric recording is enabled, we don't necessarily need to require to instantiate and pass SMR if server metrics are not collected. RR + GSLB is just one example, but I don't think it's prod specific.

If we want all of the above I think we need a way to tell if the call metric recording was enabled when the server was built. I could not find a way other than storing a flag.

Reply:
Hmm, good point about having `ExperimentalGetCallMetricRecorder()` return null if call metric reporting is not enabled.  And you're right, the only reasonable way to do that is to have another bool here -- but if we still want to allocate the `CallMetricRecorder` only when it's actually needed, then we would also need that bool in `ServerContext`, not just in the server, which would increase per-call memory.  It seems better not to do that, especially since it should be rare for an application to enable per-call reporting but then not actually send any per-call stats.

Given this trade-off, let's just give up on the idea of creating the `CallMetricRecorder` lazily.  It's fine to create it just once, the way you have it already.